projects/Mallard/src/components/weather.tsx

```diff
@@ -138,79 +143,96 @@ const WeatherIconView = ({
-                                            style={styles.forecastItemLong}
-                                            key={forecast.EpochDateTime}
-                                            forecast={forecast}
-                                        />
-                                    )
-                                })}
-                            </GridRowSplit>
-                        </View>
-                    ),
-                    [Breakpoints.tabletVertical]: () => (
-                        <View style={styles.weatherContainerNarrow}>
-                            {intervals.map(forecast => {
-                                return (
-                                    <WeatherIconView
-                                        style={styles.forecastItemNarrow}
-                                        key={forecast.EpochDateTime}
-                                        forecast={forecast}
-                                        iconSize={1.5}
-                                    />
-                                )
-                            })}
+            <View style={styles.weatherContainerLong}>
+                <GridRowSplit
+                    proxy={
+                        <View style={styles.locationNameContainer}>
+                            {showSetLocationButton ? (
+                                <Button
+                                    onPress={onSetLocationPress}
+                                    appearance={ButtonAppearance.skeleton}
+                                    style={styles.setLocationButtonWrap}
+                                    buttonStyles={styles.setLocationButton}
+                                >
+                                    Set Location
+                                </Button>
+                            ) : (
+                                <>
+                                    <Text style={styles.locationPinIcon}>
+                                        {'\uE01B'}
+                                    </Text>
+                                    <Text style={styles.locationName}>
+                                        {locationName}
+                                    </Text>
+                                </>
+                            )}
                         </View>
-                    ),
-                }}
-            </WithBreakpoints>
+                    }
+                >
+                    {intervals.map(forecast => {
+                        return (
+                            <WeatherIconView
+                                style={styles.forecastItemLong}
+                                key={forecast.EpochDateTime}
+                                forecast={forecast}
+                            />
+                        )
+                    })}
+                </GridRowSplit>
+            </View>
         )
     }
 
-    return <></>
+    return <Text>invalid weather</Text>
```

Comment:
Remove extra text. Then again it's probably bad practice not to show anything if these forecasts are "invalid". Gotta restructure to validate earlier in the chain (ex. in the resolver) so that it's a full-fledged error.